### ISSUE TITLE
Partial record check doesn't throw

### ISSUE THREAD

**Issue by gburnett** [pelotom/runtypes#188]
I want to check a record that only has optional properties.  I can define a runtype that I think is correct but neither check, nor guard, will invalidate input.  For example:

```
import { Record, Static, String } from "runtypes";

const MyRecordRuntype = Record({ foo: String }).asPartial();
const { check, guard } = MyRecordRuntype;

type MyRecord = Static<typeof MyRecordRuntype>; // { foo?: string | undefined }

// const invalidRecord: MyRecord = "Oh hai, mark!";
// Type '"Oh hai, mark!"' has no properties in common with type '{ foo?: string | undefined; }'.

const invalidRecord = "Oh hai, mark!";

const validRecord = { foo: "bar" };
const validRecord2 = {};

describe("runTypes/record", () => {
  describe("check", () => {
    it("throws for an invalid record", () => {
      expect(() => check(invalidRecord)).toThrow();
    }); // fail

    it("does not throw for a valid record", () => {
      expect(() => check(validRecord)).not.toThrow();
      expect(() => check(validRecord2)).not.toThrow();
    }); // pass
  });

  describe("guard", () => {
    it("returns false for an invalid record", () => {
      expect(guard(invalidRecord)).toBe(false);
    }); // fail

    it("returns true for a valid record", () => {
      expect(guard(validRecord)).toBe(true);
      expect(guard(validRecord2)).toBe(true);
    }); // pass
  });
});
```

The derived static type behaves as expected.  Check does not throw and guard does not return false.

Is there a correct way to define a record that only has optional properties?

I'm using `runtypes@5.0.1` and `typescript@3.8.2`.

**Comment by yuhr** [comment]
Thanks for filing the issue. I'm not sure whether we should handle this or not. At least what I can say is that `"Oh hai, mark!"` practically conforms to the type `{ foo?: string | undefined }`, but TypeScript doesn't allow it statically...

**Comment by yuhr** [comment]
https://github.com/pelotom/runtypes/issues/8#issuecomment-581044244
> I'd say when in doubt, WWTSD (what would typescript do)? Model it after how typescript reports static errors.

So we'd rather conform to TS here!

**Comment by yuhr** [comment]
It was unintended but resolved within #209 😆 This will be available in v6.
